docs/dev/stream/testing.md

```diff
@@ -332,6 +332,87 @@ Many more examples for the usage of these test harnesses can be found in the Fli
 
 <span class="label label-info">Note</span> Be aware that `AbstractStreamOperatorTestHarness` and its derived classes are currently not part of the public API and can be subject to change.
 
+### Unit Testing ProcessFunction
```

Comment:
This should be a subsection of the "Unit Testing Stateful or Timely UDFs & Custom Operators" (`####` instead of `###`) as the previous test harnesses can be used directly to test a `ProcessFunction` and the `ProcessFunctionTestHarnesses` is simply a factory with facilitation methods.